scripts/can_errgo/error_tui.go

```diff
@@ -0,0 +1,633 @@
+		switch {
+		case key.Matches(msg, m.tableKeys.q):
+			return m, tea.Quit
+		case key.Matches(msg, m.tableKeys.a):	
+			
+			// Prevent panic if no row is selected
+			if m.table.SelectedRow() == nil{
+				m.table.SetCursorAndViewport(0)
+				return m, nil
+			}
+
+			// Remove the row and reset its count.
+			errorIndex := m.table.SelectedIndex()
+			m.errorData[errorIndex].Count = 0
+			newRows := toTableRows(m.errorData)
+			m.table.SetRows(newRows)
+			return m, nil
+		case key.Matches(msg, m.tableKeys.i):
+
+			// Prevent panic if no row is selected
+			if m.table.SelectedRow() == nil{
+				m.table.SetCursorAndViewport(0)
+				return m, nil
+			}
+
+			// Toggle the mask for the selected error and move it to the ignore menu.
+			errorIndex := m.table.SelectedIndex()
+			m.ignoreMask = toggleBit(m.ignoreMask, errorIndex)
+			m.errorData[errorIndex].Ignored = true
+
+			// Remove the row from the main table and reset its count.
+			m.errorData[errorIndex].Count = 0
+			newRows := toTableRows(m.errorData)
+			m.table.SetRows(newRows)
+
+			//Set add the ignored rows to submenu
+			ignoredRows := m.getIgnoredRows()
+			m.submenuTable.SetRows(ignoredRows)
+
+			return m, nil
+        case key.Matches(msg, m.tableKeys.s):
+            if m.submenuActive {
+                // Hide submenu and return to main menu
+                m.submenuActive = false
+                m.table.Focus()
+            } else {
+                // Show submenu and focus it
+                m.submenuActive = true
+                m.table.Blur()
+                m.submenuTable.Focus()
+				m.table.SetShowCursor(false)
+            }
+		case key.Matches(msg, table.DefaultKeyMap().LineUp):
+			m.table.MoveUp(1)
+			return m,nil	
+		case key.Matches(msg, table.DefaultKeyMap().LineDown):
+			m.table.MoveDown(1)
+			return m,nil			
+	}
+	case CANMsg:
+		// Reset the timeout flag and update the last CAN message time
+		m.lastCANTime = time.Now()
+		m.isTimeout = false
+		// Perform bitwise AND to filter out the bits that are not of interest
+		msg.Value = msg.Value & m.ignoreMask
+		for i,errorVal := range m.errorData{
+			m.errorData[i].Recency += 1
+			if msg.Value&(1<<errorVal.ErrorIndex) != 0{
+				m.errorData[i].Count += 1
+				m.errorData[i].Recency = 0
+			}
+		}
+		// Sort the table by the recency.
+		currentError := m.table.SelectedIndex()
+		newRowValues :=  SortRowValuesByRecency(m.errorData)
+		newRows := toTableRows(newRowValues)
+		m.table.SetRows(newRows)
+
+		// Fix the cursor back onto the row it was on before the sort.
+		if currentError != -1{
+			newIndex := findError(currentError, newRowValues)
+			m.table.SetCursorAndViewport(newIndex)
+		}
+		return m, nil
+	// If the message is a tick, check if the last CAN message was received more than the timeout duration ago
+	case TickMsg:
+		if time.Since(m.lastCANTime) > m.t {
+			m.isTimeout = true
+		}
+		return m, tickEvery(m.t)
+	}
+
+	// Update the table with the new message
+	if _, ok := msg.(CANMsg); ok {
```

Comment:
what is this condition checking? It seems to only update the main menu in one case and the submenu in the other, contrary to the next comment